File: 0561-array-partition/0561-array-partition.py
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        nums.sort()

        maxSum = 0 

        for i in range(0, len(nums), 2):
            maxSum += nums[i]
        
        return maxSum

# Sorting and skipping even numbers
#time Complexity: O(nlogn)
# Space Complexity: O(1)

File: 0561-array-partition/0561-array-partition.java
class Solution {
    public int arrayPairSum(int[] nums) {
        if(nums == null || nums.length == 0){
            return 0;
        }
        Arrays.sort(nums);
        int n = nums.length;

        int minSum = 0;

        for(int i = 0; i < n; i += 2){
            minSum += nums[i];
        }

        return minSum;
    }
}


//Sorting
// Time Compplexity:O(nlogn)
// Space Complexity: O(1)